src/tir/analysis/block_access_region_detector.cc

```diff
@@ -170,12 +172,12 @@ void BlockReadWriteDetector::VisitStmt_(const IfThenElseNode* op) {
   VisitExpr(op->condition);
   {
     // Visit then branch
-    With<ConditionalBoundsContext> ctx(op->condition, &dom_map_, &hint_map_, true);
+    With<ConditionalBoundsContext> ctx(op->condition, &dom_map_, &hint_map_, &pending_conditions_);
```

Comment:
Should these `With<ConditionalBoundsContext>` also be applied to a `BlockRealizeNode::predicate`?

Reply:
yes! it is applied when we visit block realize